### PR TITLE
Improve embed feature

### LINKED ISSUE #1938: Improve Embed Feature.
Currently the embed feature is basic, it just embeds under some default settings. Would like something as follows:

1. Click Embed
2. Dialog Opens
3. Dialog has following options

- iFrame Size
- Color Theme
- Optional Project Title Display
- Landing Circuit Tab (Circuit that is displayed be default)  
- Some Styling options like border etc
- Optionally Disable stuff like Clock Enable/Clock Time Interval/Zoom/FullScreen options 
- Some more stuff like this 

Referencing https://github.com/CircuitVerse/CircuitVerse/issues/1555

Current Sample iFrame Code
```
<iframe width="600px" height="400px" 
src="https://circuitverse.org/simulator/embed/sap-1-computer-5a0acd53-6e0c-434d-b19d-4073d9a6dc58" 
id="projectPreview" scrolling="no" 
webkitAllowFullScreen mozAllowFullScreen allowFullScreen></iframe>
```

I think we can encode the options like this 

`https://circuitverse.org/simulator/embed/<project id>?color-theme="default"&landing-circuit-id="21323"&display-title="true"`

@danielluehr would like your inputs here. 
@sal2701 @Shivansh2407 @nitin10s

### PR DESCRIPTION
**Pull request by Kartikay304**
fixes #1938

#### Describe the changes you have made in this PR -
improved embed feature, added border-width&color and iframe-width&height

- [x] Iframe width and height.
- [x] Border width and color.
- [x] Optionally Disable stuff like Clock Enable/Clock Time Interval/Zoom
- [x] Change theme

### Screenshots of the changes (If any) -

Screenshot from 2022-06-28 14-37-43

https://user-images.githubusercontent.com/76901313/176727876-ffc23620-d23a-464a-9c14-2486017e4a10.mp4

### CODE REVIEW

**Comment by Kartikay304** [comment]
@tachyons @nitin10s @gr455  Kindly review my PR.

**Comment by vedant-jain03** [comment]
@Kartikay304 Can we increase the margin of textarea.

**Comment by Kartikay304** [comment]
> @Kartikay304 Can we increase the margin of textarea.

Sure we can increase the margin around the text area but, I intentionally designed the text area this way, lets wait for Nitin's review for improvements in design.

**Comment by gr455** [comment]
Can we put default values on the embed modal (say width 500, height 500). So that someone can just quickly generate an embed

**Comment by gr455** [comment]
> @tachyons @nitin10s @gr455 Sir, can you please take a look at my PR.

Also you don't need to call anybody 'sir' haha :sweat_smile:

File: app/views/projects/show.html.erb [hunk 1]
```diff
@@ -141,19 +141,69 @@
...
           <% end %>
-          <div class="collapse" id="collapseExample">
-            <div class="projects-embed-text-container">
-              <label for="embedTextArea"></label>
-              <textarea id="embedTextArea" class="projects-embed-text" rows="4" cols="35"><iframe width="600px" height="400px" src="<%= simulator_embed_url(@project) %>" id="projectPreview" scrolling="no" webkitAllowFullScreen mozAllowFullScreen allowFullScreen></iframe></textarea>
+                <div id="embed"
+                  class="modal fade rounded "
+                  role="dialog" tabindex="-1"
+                  role="dialog"
+                  aria-labelledby="mySmallModalLabel"
```
Comment (tachyons) [inline app/views/projects/show.html.erb:154]: Update the label here

File: app/views/projects/show.html.erb [hunk 2]
```diff
@@ -267,6 +305,19 @@
   }
 </script>
 
+<script>
+  function edit_embed(){
```
Comment (tachyons) [inline app/views/projects/show.html.erb:309]: Can we use stimulus-js here to cleanup the implementation ?

**Comment by vedant-jain03** [comment]
@Kartikay304 Thank you for your valuable contribution. This PR needs to be completed ASAP and is now considered under GSoC22 Platform Improvement Project. I am going to work on it from now!

Thankyou 🎉

**Comment by vedant-jain03** [comment]
@tachyons @nitin10s @pavanjoshi914 @abstrekt Kindly review.

File: app/javascript/controllers/projects_controller.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import { Controller } from 'stimulus';
+
+export default class extends Controller {
+    handleMainCheckbox() {
+        $('#advance-embed').change((e) => {
```
[the remark this answers is not in the document]
Comment (vedant-jain03) [reply app/javascript/controllers/projects_controller.js:5]: Expected

File: app/component/project_embed_component.html.erb [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+                <input class="form-check-input element-restriction" type="checkbox" id="checkbox-fullscreen" value="Fullscreen" checked="true">
+                <div class="primary-checkpoint"></div>
+                <h6>Fullscreen</h6>
+              </label>
+            </div>
+            <div class="form-check form-check-inline">
+              <label class="form-check-label primary-checkpoint-container">
+                <input class="form-check-input element-restriction" type="checkbox" id="checkbox-zoom-in-out" value="Zoom In Out" checked="true">
+                <div class="primary-checkpoint"></div>
+                <h6>Zoom In/Out</h6>
```
Comment (tachyons) [inline app/component/project_embed_component.html.erb:73]: These texts need to translatable

**Review by gr455 — CHANGES_REQUESTED** [review]
Disabling zoom seems to not work (tested on chrome lastest)

**Comment by vedant-jain03** [comment]
> Disabling zoom seems to not work (tested on chrome lastest)

https://user-images.githubusercontent.com/76901313/176892332-9f03f6b8-c2a8-41cc-aaae-ef8172752441.mp4

@gr455 have a look at this clip, is that the same as what you did for disabling the zoom-in-out?

File: app/component/project_embed_component.html.erb [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+                  <option value="color-blind">Color Blind</option>
+              </select>
+            </div>
+          </div>
+        </div>
+        <div>
+            <button class="btn primary-button projects-primary-button embed-button" data-action="click->projects#showEmbedLink"><%= t("projects.modal.publish_button") %></button>
+        </div>
+        <div class="collapse show">
+            <textarea name='embed-textarea' class="form-control form-input embed-link-textarea" rows="5px" column="115px" id="result" readonly></textarea>
```
[the remark this answers is not in the document]
Comment (vedant-jain03) [reply app/component/project_embed_component.html.erb:93]: It is readonly textarea, Not Needed!

File: app/component/project_embed_component.html.erb [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
```
[the remark this answers is not in the document]
Comment (vedant-jain03) [reply app/component/project_embed_component.html.erb:93]: It is readonly textarea, Not Needed!

File: app/javascript/controllers/projects_controller.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
+/* eslint-disable class-methods-use-this */
+import { Controller } from 'stimulus';
+
+var flag = false;
+
+export default class extends Controller {
+    handleMainCheckbox() {
+        document.querySelector('.advance-embed-option').style.display = (flag) ? 'none' : 'block';
+        document.querySelector('.advance-embed-field .dropdown').style.transform = (flag) ? 'rotate(0deg)' : 'rotate(180deg)';
+        flag = !flag;
```
Comment (tachyons) [inline app/javascript/controllers/projects_controller.js:10]: if you are using this for toggler, you can simplify by using https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details

**Comment by tachyons** [comment]
Great job @vedant-jain03 and @Kartikay304 🎈

**Pull request merged**